tonic/src/codec/mod.rs

```diff
@@ -21,6 +21,41 @@ pub use self::decode::Streaming;
 #[cfg(feature = "prost")]
 #[cfg_attr(docsrs, doc(cfg(feature = "prost")))]
 pub use self::prost::ProstCodec;
+#[cfg(feature = "prost")]
+#[cfg_attr(docsrs, doc(cfg(feature = "prost")))]
+pub use self::prost::ProstDecoder;
+#[cfg(feature = "prost")]
+#[cfg_attr(docsrs, doc(cfg(feature = "prost")))]
+pub use self::prost::ProstEncoder;
+
+/// Unless overridden, this is the buffer size used for encoding requests.
+/// This is spent per-rpc, so you may wish to adjust it. The default is
+/// pretty good for most uses, but if you have a ton of concurrent rpcs
+/// you may find it too expensive.
+const DEFAULT_CODEC_BUFFER_SIZE: usize = 8 * 1024;
+const DEFAULT_YIELD_THRESHOLD: usize = 32 * 1024;
+
+/// Settings for how tonic allocates and grows buffers.
+#[derive(Clone, Copy, Debug)]
+pub struct BufferSettings {
+    /// Initial buffer size, and the growth unit for cases where the size
+    /// is larger than the buffer's current capacity. Defaults to 8 KiB.
+    ///
+    /// Notably, this is eagerly allocated per streaming rpc.
+    pub buffer_size: usize,
```

Comment:
I would like to avoid exposing public struct fields like this and would rather have setters/getters for them or a new fn that accepts them. I would then move the field docs to the struct as well.

Reply:
Of course, I'll adjust that. Thanks